java-generator/core/src/main/java/io/fabric8/java/generator/nodes/AbstractJSONSchema2Pojo.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.nodes;
+
+import static io.fabric8.java.generator.nodes.Keywords.JAVA_KEYWORDS;
+
+import com.github.javaparser.ast.CompilationUnit;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+
+public abstract class AbstractJSONSchema2Pojo {
+
+    public abstract String getType();
+
+    public abstract GeneratorResult generateJava(CompilationUnit cu);
+
+    public static String sanitizeString(String str) {
+        String sanitized = "";
+        if (JAVA_KEYWORDS.stream().filter((s) -> s.equals(str)).findFirst().isPresent()) {
+            sanitized = "_" + str;
+        } else {
+            sanitized = str;
+        }
+
+        if (sanitized.startsWith("-")) {
+            sanitized = sanitized.replaceFirst("-", "minus");
+        }
+
+        int index = sanitized.indexOf('-');
+        while (index != -1) {
+            int next = Math.min(index + 2, sanitized.length());
+            sanitized =
+                    sanitized.substring(0, index)
+                            + sanitized.substring(index + 1, next).toUpperCase()
+                            + sanitized.substring(next);
+            index = sanitized.indexOf('-');
+        }
+
+        return sanitized;
+    }
+
+    public static AbstractJSONSchema2Pojo fromJsonSchema(
+            String key, JSONSchemaProps prop, String prefix, String suffix) {
+        if (prop.getXKubernetesIntOrString() != null && prop.getXKubernetesIntOrString()) {
+            return fromJsonSchema(
+                    key,
+                    new JPrimitiveNameAndType("io.fabric8.kubernetes.api.model.IntOrString"),
+                    prop,
+                    prefix,
+                    suffix);
+        } else if (prop.getType() == null
+                && prop.getXKubernetesPreserveUnknownFields() != null
+                && prop.getXKubernetesPreserveUnknownFields()) {
+            return fromJsonSchema(key, new JObjectNameAndType(key), prop, prefix, suffix);
+        } else if (prop.getEnum() != null && prop.getEnum().size() > 0) {
+            return fromJsonSchema(key, new JEnumNameAndType(key), prop, prefix, suffix);
+        } else {
+            if (prop.getType() == null) {
+                throw new RuntimeException("Type for key:" + key + " is null");
```

Comment:
Probably should use more specific exception like `IllegalArgumentException` maybe?

Reply:
👍